core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedScheduleGenerator.scala

```diff
@@ -182,9 +204,18 @@ class CostBasedScheduleGenerator(
       )
     }
     // Since the plan is now schedulable, calling the search directly returns a region DAG.
+    // This new search is only needed because of additional cache read operators.
+    // As a result the original materialized ports are not kept in the new region DAG.
+    // TODO: remove this step after cache read is removed.
```

Comment:
I don't understand this comment, and also why we need to remove this search when cache read is removed?

Reply:
This second search is only used to rebuild the regionDAG since adding materializations using cache read operators changes the physical plan and requires the new cache read operators to be in the regions. We will remove sink operators soon, and once cache read operators are also removed, this second search will not be needed, and the result regionDAG of the first search will be the final region DAG. I included more detailed explanations in the comments.